teps/0044-decouple-task-composition-from-scheduling.md

```diff
@@ -14,113 +14,138 @@ authors:
+This issue provides more detail on why it's difficult to share data between parallel tasks using PVCs.
+- [Feature Request: Pooled PersistentVolumeClaims](https://github.com/tektoncd/pipeline/issues/3417):
+This user would like to attach preallocated PVCs to PipelineRuns and TaskRuns rather than incurring the overhead of creating a new one every time.
+- [@mattmoor's feedback on PipelineResources and the Pipeline beta](https://twitter.com/mattomata/status/1251378751515922432):
+The experience of running a common, fundamental workflow is made more difficult by having to use PVCs to move data between pods.
+- [Issue: Exec Steps Concurrent in task (task support DAG)](https://github.com/tektoncd/pipeline/issues/3900): This user would like to be able to
+run Task Steps in parallel, because they do not want to have to use workspaces with multiple pods.
+- [Another comment on the previous issue](https://github.com/tektoncd/pipeline/issues/3900#issuecomment-848832641) from a user who would like to be
+able to run Steps in parallel, but doesn't feel that running a Pipeline in a pod would address this use case because they don't want to turn their Steps into Tasks.
+- [Question: without using persistent volume can i share workspace among tasks?](https://github.com/tektoncd/pipeline/issues/3704#issuecomment-980748302):
+This user uses an NFS for their workspace to avoid provisioning a PVC on every PipelineRun.
+- [FR: Allow volume from volumeClaimTemplate to be used in more than one workspace](https://github.com/tektoncd/pipeline/issues/3440):
+This issue highlights usability concerns with using the same PVC in multiple workspaces (done using sub-paths).
+
+## Existing Workarounds and Mitigations
+
+There's currently no workaround that addresses the overhead of extra pods or storage without harming reusability.
+
+### Combine multiple pieces of functionality in one Task
+Instead of combining functionality provided by Tasks into Pipelines, a Task or Pipeline author could use Steps or a multifunctional script to combine
+all necessary functionality into a single Task. This allows multiple "actions" to be run in one pod, but hurts reusability and makes parallel execution more difficult.
+
+### Use PipelineResources (deprecated) to express a workflow in one Task
+PipelineResources allowed multiple pieces of functionality to run in a single pod by building some of these functions into the TaskRun controller.
+This allowed some workflows to be written as single Tasks.
+For example, the "git" and "image" PipelineResources made it possible to create a workflow that cloned and built a repo, and pushed the resulting image to
+an image repository, all in one Task. 
+However, PipelineResources hurt reusability because they required Task authors to anticipate what other functionality would be needed before and after the Task.
+For this reason, among others, they were deprecated; please see [TEP-0074](./0074-deprecate-pipelineresources.md) for more information.
```

Comment:
one more detail: pipelineresources address the multiple pod problem but not the storage problem - even if it could be done more automagically, using pipelineresources to shuttle data between tasks still involved either an underlying PVC or for the data to be moved to and from external storage between taskruns (https://github.com/tektoncd/pipeline/blob/main/docs/install.md#configuring-pipelineresource-storage)

Reply:
good point thanks!